src/app/src/main/kotlin/org/imperial/mrc/hint/exceptions/ExceptionHandler.kt

```diff
@@ -18,21 +19,22 @@ import javax.validation.ConstraintViolationException
 
 @Order(Ordered.HIGHEST_PRECEDENCE)
 @ControllerAdvice
-class HintExceptionHandler : ResponseEntityExceptionHandler() {
+class HintExceptionHandler(private val errorCodeGenerator: ErrorCodeGenerator,
+                           private val appProperties: AppProperties) : ResponseEntityExceptionHandler() {
 
     override fun handleExceptionInternal(e: java.lang.Exception,
                                          @Nullable body: Any?,
                                          headers: HttpHeaders,
                                          status: HttpStatus,
                                          request: WebRequest): ResponseEntity<Any> {
         logger.error(e.message)
```

Comment:
So this ExceptionHandler is using its inherited logger to log out any exception messages anyway. Maybe this is redundant now? Maybe the Filter should be using a configured logger rather than defaulting to system log?

Reply:
I guess a log factory logger might be better for the filter, so that the error messages are formatted the same throughout the app. Not too worried about redundancy and in some instances this is logging something different to the filter: https://github.com/mrc-ide/hint/pull/168/files#diff-2a06c226eeb55efc9a48ced0a230f53bL46